src/main/java/com/shapesecurity/salvation2/Directives/SandboxDirective.java

```diff
@@ -29,327 +56,146 @@ public SandboxDirective(List<String> values, DirectiveErrorConsumer errors) {
-					}
-					break;
-				case "allow-pointer-lock":
-					if (!this.allowPointerLock) {
-						this.allowPointerLock = true;
-					} else {
-						errors.add(Policy.Severity.Warning, "Duplicate sandbox keyword allow-pointer-lock", index);
-					}
-					break;
-				case "allow-popups":
-					if (!this.allowPopups) {
-						this.allowPopups = true;
-					} else {
-						errors.add(Policy.Severity.Warning, "Duplicate sandbox keyword allow-popups", index);
-					}
-					break;
-				case "allow-popups-to-escape-sandbox":
-					if (!this.allowPopupsToEscapeSandbox) {
-						this.allowPopupsToEscapeSandbox = true;
-					} else {
-						errors.add(Policy.Severity.Warning, "Duplicate sandbox keyword allow-popups-to-escape-sandbox", index);
-					}
-					break;
-				case "allow-presentation":
-					if (!this.allowPresentation) {
-						this.allowPresentation = true;
-					} else {
-						errors.add(Policy.Severity.Warning, "Duplicate sandbox keyword allow-presentation", index);
-					}
-					break;
-				case "allow-same-origin":
-					if (!this.allowSameOrigin) {
-						this.allowSameOrigin = true;
-					} else {
-						errors.add(Policy.Severity.Warning, "Duplicate sandbox keyword allow-same-origin", index);
-					}
-					break;
-				case "allow-scripts":
-					if (!this.allowScripts) {
-						this.allowScripts = true;
-					} else {
-						errors.add(Policy.Severity.Warning, "Duplicate sandbox keyword allow-scripts", index);
-					}
-					break;
-				case "allow-storage-access-by-user-activation":
-					if (!this.allowStorageAccessByUserActivation) {
-						this.allowStorageAccessByUserActivation = true;
-					} else {
-						errors.add(Policy.Severity.Warning, "Duplicate sandbox keyword allow-storage-access-by-user-activation", index);
-					}
-					break;
-				case "allow-top-navigation":
-					if (!this.allowTopNavigation) {
-						this.allowTopNavigation = true;
-					} else {
-						errors.add(Policy.Severity.Warning, "Duplicate sandbox keyword allow-top-navigation", index);
-					}
-					break;
-				case "allow-top-navigation-by-user-activation":
-					if (!this.allowTopNavigationByUserActivation) {
-						this.allowTopNavigationByUserActivation = true;
-					} else {
-						errors.add(Policy.Severity.Warning, "Duplicate sandbox keyword allow-top-navigation-by-user-activation", index);
-					}
-					break;
-				default:
-					if (token.startsWith("'")) {
-						errors.add(Policy.Severity.Error, "Unrecognized sandbox keyword " + token + " - note that sandbox keywords do not have \"'\"s", index);
-					} else {
-						errors.add(Policy.Severity.Error, "Unrecognized sandbox keyword " + token, index);
-					}
+			Value value = Value.fromString(lowcaseToken);
+			if(value == null) {
+				if (token.startsWith("'")) {
+					errors.add(Policy.Severity.Error, "Unrecognized sandbox keyword " + token + " - note that sandbox keywords do not have \"'\"s", index);
```

Comment:
If the error message is going to state double quote single quote shouldn't the condition also check double quote?

`if (token.startsWith("'") || token.startsWith("\"")) {`

Reply:
Example of this error message would be:
```
Unrecognized sandbox keyword 'strict-dynamic' note that sandbox keywords do not have "'"s
```
I have left this part intact, but I think the reason of mentioning just single quote is the fact, that Content Security Policy allows them in some situations. For [keywords in Source List](https://w3c.github.io/webappsec-csp/#grammardef-keyword-source) must have single quotes, they are handled in [SourceExpressionDirective](https://github.com/shapesecurity/salvation/blob/main/src/main/java/com/shapesecurity/salvation2/Directives/SourceExpressionDirective.java).

I guess we could include `"` in the condition and also change message to `shouldn't be surrounded with quotes`. That would be ok?

Reply:
Alright, I have added it to PR.